extensions/music/src/util/scripts/playlists.ts

```diff
@@ -56,3 +59,38 @@ export const getPlaylists = (kind: PlaylistKind): TE.TaskEither<Error, string> =
         end tell
 	return output
 `);
+
+export const getArtworkByIds = async (ids: string[]) => {
+  const result: any = {};
+  const layout = getViewLayout();
+  const size = getImageSize();
+  const num = ids.length;
+  const promises = ids.map(async (id) => {
+    const res = await runAppleScript(`
+      tell application "Music"
+        set playlistArtwork to first artwork of first playlist of (every playlist whose id is ${id})
+        set playlistImage to data of playlistArtwork
+      end tell
+      return playlistImage
+    `);
+    let binary = null;
+    try {
```

Comment:
Everything seems ok to me. The only thing I would do differently is by wrapping the following try/catch into a separate generic function (since the code seems to be almost identical in 3 files.)

Reply:
Sounds good, I'll add this soon.